cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -2460,6 +2461,558 @@ TEST(GroupBy, Distinct) {
   }
 }
 
+MATCHER_P(AnyOfScalar, arrow_array, "") {
+  for (int64_t i = 0; i < arrow_array->length(); ++i) {
+    auto scalar = arrow_array->GetScalar(i).ValueOrDie();
```

Comment:
We could handle the error instead and report an assertion failure if GetScalar fails.